BridgeSDK/BridgeAPI/BridgeAPITypes/MachineGeneratedDoNotEdit/_SBBActivity.m

```diff
@@ -89,23 +129,190 @@ - (NSDictionary *)dictionaryRepresentation
+    [self updateManagedObject:managedObject withObjectManager:objectManager cacheManager:cacheManager];
+
+    // Calling code will handle saving these changes to cacheContext.
+
+    return managedObject;
+}
+
+- (NSManagedObject *)saveToContext:(NSManagedObjectContext *)cacheContext withObjectManager:(id<SBBObjectManagerProtocol>)objectManager cacheManager:(id<SBBCacheManagerProtocol>)cacheManager
+{
+    NSManagedObject *managedObject = [cacheManager cachedObjectForBridgeObject:self inContext:cacheContext];
+    if (managedObject) {
+        [self updateManagedObject:managedObject withObjectManager:objectManager cacheManager:cacheManager];
+    }
+
+    // Calling code will handle saving these changes to cacheContext.
+
+    return managedObject;
+}
+
+- (void)updateManagedObject:(NSManagedObject *)managedObject withObjectManager:(id<SBBObjectManagerProtocol>)objectManager cacheManager:(id<SBBCacheManagerProtocol>)cacheManager
+{
+
+    [super updateManagedObject:managedObject withObjectManager:objectManager cacheManager:cacheManager];
+    NSManagedObjectContext *cacheContext = managedObject.managedObjectContext;
+
+    managedObject.activityType = ((id)self.activityType == [NSNull null]) ? nil : self.activityType;
+
+    managedObject.guid = ((id)self.guid == [NSNull null]) ? nil : self.guid;
+
+    managedObject.label = ((id)self.label == [NSNull null]) ? nil : self.label;
+
+    managedObject.labelDetail = ((id)self.labelDetail == [NSNull null]) ? nil : self.labelDetail;
+
+    // destination entity SurveyReference is not directly cacheable, so delete it and create the replacement
+    if (managedObject.survey) {
+        [cacheContext deleteObject:managedObject.survey];
+    }
+    NSManagedObject *relMoSurvey = [self.survey createInContext:cacheContext withObjectManager:objectManager cacheManager:cacheManager];
+
+    [managedObject setSurvey:relMoSurvey];
+
+    // destination entity SurveyResponseReference is not directly cacheable, so delete it and create the replacement
+    if (managedObject.surveyResponse) {
+        [cacheContext deleteObject:managedObject.surveyResponse];
+    }
+    NSManagedObject *relMoSurveyResponse = [self.surveyResponse createInContext:cacheContext withObjectManager:objectManager cacheManager:cacheManager];
+
+    [managedObject setSurveyResponse:relMoSurveyResponse];
+
+    // destination entity TaskReference is not directly cacheable, so delete it and create the replacement
+    if (managedObject.task) {
+        [cacheContext deleteObject:managedObject.task];
+    }
+    NSManagedObject *relMoTask = [self.task createInContext:cacheContext withObjectManager:objectManager cacheManager:cacheManager];
+
+    [managedObject setTask:relMoTask];
+
+    // Calling code will handle saving these changes to cacheContext.
+}
+
 #pragma mark Direct access
 
+- (void) setSurvey: (SBBSurveyReference*) survey_ settingInverse: (BOOL) setInverse
```

Comment:
This is generated by the template and is meant to handle keeping the inverse relationship connected properly in the PONSO (Plain Old NSObjects) generated code. Since I don't want those inverse relationships to show up in the dictionary/Bridge JSON that gets generated from these objects, I added a flag "notInPONSODictionary" in the userInfo on these relationships, and had it also prevent that code from being generated. We don't use it anyway and the methods with that parameter aren't publicly available.